### ISSUE TITLE
`--createstub` can generate stub files with missing imports

### ISSUE THREAD

**Issue by Kimundi** [microsoft/pyright#2975]
**Describe the bug**

The generated stub files for a module might not contain imports needed by some definitions in it. See example.

**To Reproduce**
1. Create an `test_module` python module consisting of the two file in the Code section.
2. Run `pyright --createstub test_module`
3. Look at the generated typestubs, and see that `typings/test_module/sub_file.pyi` does not contain an import for `urllib.parse`

**Expected behavior**

`typings/test_module/sub_file.pyi` should contain `import urllib.parse`, so that the `Export = urllib.parse.ParseResult` line references an name that is imported.

**Code**
***Example module***

```python
# test_module/__init__.py
from .sub_file import Export
```

```python
# test_module/sub_file.py
import urllib.parse

Export = urllib.parse.ParseResult
```

***Generated type stubs for this module***

```python
# typings/test_module/__init__.pyi
"""
This type stub file was generated by pyright
"""

from .sub_file import Export
```

```python
# typings/test_module/sub_file.pyi
"""
This type stub file was generated by pyright
"""

Export = urllib.parse.ParseResult
```

**VS Code extension or command-line**
Running from the commandline
```
> pyright --version
pyright 1.1.217
```

**Comment by erictraut** [comment]
Thanks for the bug report. The stub generation code attempts to omit import statements that are not referenced, but this logic contained a bug where it wasn't properly handling references to multi-part imports like `import urllib.parse`. A fix will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.218, which I just published. It will also be included in the next release of pylance.
